warehouse/query-core/src/main/java/datawave/query/iterator/logic/AndIterator.java

```diff
@@ -379,14 +396,23 @@ protected TreeMultimap<T,NestedIterator<T>> advanceIterators(T key) {
                     highest = transform;
                 }
             } catch (Exception e) {
-                // only need to actually fail if we have nothing left in the AND clause
-                if (includeHeads.isEmpty()) {
-                    throw e;
+                seenException = true;
+                if (itr.isNonEventField()) {
+                    // dropping a non-event term from the query means that the accuracy of the query
```

Comment:
Do you mean an event only field here?  We can certainly lookup non-event fields in the index so I am a little confused.  What am I missing?

Reply:
We *can* look up non-event or index-only fields in the index, but only under certain cases. 

The DelayedNonEventIndexContext only looks for non-event fields within delayed expressions, which by definition do not contribute iterators to the AndIterator.

The IndexOnlyContextCreator only created an index only context if the query contains the special marker "@LAZY_SET_FOR_INDEX_ONLY_FUNCTION_EVALUATION". 

The case we're dealing with here is `INDEXED_EVENT_FIELD == 'foo' && INDEX_ONLY_FIELD == 'bar'`. If INDEXED_EVENT_FIELD is dropped from the intersection we can still drive the query with INDEX_ONLY_FIELD because we will pick up the INDEXED_EVENT_FIELD while aggregating the event.

However, if we drop the INDEX_ONLY_FIELD there's no way to get that back into the document.

Reply:
I am so sorry.  I missed the context if the check here entirely.  You are correct.